src/recipe/recipe.service.ts

```diff
@@ -105,18 +105,45 @@ export class RecipeService {
     return new RecipeResponse(recipeFromDb);
   }
 
-  async findAllRecipes(limit: number, page: number): Promise<RecipeResponse[]> {
-    const recipesFromDb = await this.prisma.recipe.findMany({
-      take: limit,
-      skip: (page - 1) * limit,
-      include: {
-        preparing: true,
-        ingredients: true,
-      },
-    });
-    return recipesFromDb.map((recipe) => {
-      return new RecipeResponse(recipe);
-    });
+  async findAllRecipes(
+    limit: number,
+    page: number,
+    name: string,
+  ): Promise<RecipeResponse[]> {
+    if (!limit) {
+      limit = 3;
+    }
+
+    if (!page) {
+      page = 1;
+    }
+
+    if (name) {
+      const recipesFromDb = await this.prisma.recipe.findMany({
+        where: { name: { contains: name } },
+        skip: (page - 1) * limit,
+        take: limit,
+        include: {
+          preparing: true,
+          ingredients: true,
+        },
+      });
+      return recipesFromDb.map((recipe) => {
+        return new RecipeResponse(recipe);
+      });
+    } else {
+      const recipesFromDb = await this.prisma.recipe.findMany({
+        skip: (page - 1) * limit,
+        take: limit,
+        include: {
+          preparing: true,
+          ingredients: true,
+        },
+      });
+      return recipesFromDb.map((recipe) => {
+        return new RecipeResponse(recipe);
+      });
```

Comment:
this is not a good practice to do it like this. Note that the only difference between both if blocks is this `where: { name: { contains: name } },` statement. 
You could easily do something like this

`const whereCodition = name ? { name: { contains: name } } : {}`

and use it in `findMany` call


https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator

https://www.techtarget.com/whatis/definition/DRY-principle#:~:text=The%20DRY%20(don't%20repeat,something%20once%2C%20and%20only%20once.